cmd/mt-parrot/generate.go

```diff
@@ -0,0 +1,63 @@
+package main
+
+import (
+	"fmt"
+	"github.com/grafana/metrictank/clock"
+	"github.com/grafana/metrictank/schema"
+	log "github.com/sirupsen/logrus"
+	"sync/atomic"
+)
+
+func produceTestMetrics(schemas []*schema.MetricData) {
+	for tick := range clock.AlignedTickLossless(testMetricsInterval) {
+		for _, metric := range schemas {
+			metric.Time = tick.Unix()
+			metric.Value = float64(tick.Unix())
+		}
+		publisher.Flush(schemas)
+		atomic.StoreInt64(&lastPublish, tick.Unix())
+		log.Infof("flushed schemas for ts %d", tick.Unix())
+	}
+}
+
+//generateSchemas generates a MetricData that hashes to each of numPartitions partitions
+func generateSchemas(numPartitions int32) []*schema.MetricData {
+	var metrics []*schema.MetricData
+	for i := int32(0); i < numPartitions; i++ {
+		metrics = append(metrics, generateSchema(i))
+	}
+	return metrics
+}
+
+//generateSchema generates a single MetricData that hashes to the given partition
+func generateSchema(desiredPartition int32) *schema.MetricData {
+	metric := schema.MetricData{
+		OrgId:    orgId,
+		Unit:     "partyparrots",
+		Mtype:    "gauge",
+		Interval: int(testMetricsInterval.Seconds()),
+	}
+
+	for i := 1; true; i++ {
+		metric.Name = fmt.Sprintf("parrot.testdata.%d.generated.%s", desiredPartition, generatePartitionSuffix(i))
```

Comment:
`generated` is implied. I rather add a specfication of what the data symbolizes / how it's generated.
e.g. identity -> y=x
in the future we may add other test cases like sine, etc
```suggestion
		metric.Name = fmt.Sprintf("parrot.testdata.%d.identity.%s", desiredPartition, generatePartitionSuffix(i))
```

Reply:
f9d103b7b354dec2a036e9f40b1f0313afa295bd